fortran/micm.F90

```diff
@@ -135,10 +135,14 @@ end function get_user_defined_reaction_rates_ordering_c
   type :: micm_t
     type(mappings_t), pointer :: species_ordering => null()
     type(mappings_t), pointer :: user_defined_reaction_rates => null()
-    type(c_ptr), private         :: ptr = c_null_ptr
+    type(c_ptr), private      :: ptr = c_null_ptr
+    integer,     private      :: number_of_grid_cells = 0
+    integer,     private      :: solver_type = 0
```

Comment:
Would it make more sense to set them to a default value rather than an invalid value? My main concern is that `solver_type` is an enum, and we expect the input to be a string that maps to the corresponding integer value. Setting them as integers could give an impression that the type is actually an integer. 
```Fortran
  enum, bind(c)
    enumerator :: Rosenbrock                 = 1
    enumerator :: RosenbrockStandardOrder    = 2
    enumerator :: BackwardEuler              = 3
    enumerator :: BackwardEulerStandardOrder = 4
  end enum
```

Reply:
good point! I added an `UndefinedSolver` option for the initial value